### PR TITLE
fix: Make suggestions component dynamic

### PR DESCRIPTION
**Pull request by CIPHERTron**
#### As of now, the suggestions component is hardcoded. This PR will make it dynamic i.e. according to the weather type of a location, the suggestions will be displayed 🚀 
This is a WIP because we still have to give a matching `type` of the weather fetched from the API in `WeatherSuggestions/constants/Items.js`

You can find the refactored Items.js [here](https://github.com/CIPHERTron/prep-project-jan-2022/blob/suggestions/src/components/WeatherSuggestions/constants/Items.js)

---

https://user-images.githubusercontent.com/56754747/152327590-5a14bcf9-a938-4c14-96d0-9add011163c8.mov

### CODE REVIEW

**Comment by CIPHERTron** [comment]
This PR is good for review. I've added a recording of this feature. It would be great if anyone can test it locally and confirm it because sometimes the `results` prop gets passed even before we get the API response which causes things to break :/

cc: @ditsuke @KshitijDarekar @Dhruv-Sachdev1313 @harshsharma6401 @Chukwuemeka-Mba (if anyone's available 😅 )

**Review by KshitijDarekar — COMMENTED** [review]
Its breaking locally .
Screenshot 2022-02-03 at 5 08 55 PM

File: src/components/WeatherSuggestions/Suggestions.jsx [hunk 1]
```diff
@@ -30,21 +29,19 @@ export default function WeatherSuggestions(results) {
...
         className="carousel"
       >
-        {/* Traverse through the items map */}
-        {Object.keys(items).length > 0 &&
-          Object.keys(items).map((item) => (
-            <div key={item} className="required-item">
-              <LottieControl animationData={items[item]} />
-            </div>
-          ))}
+        {items.animations.map((item, idx) => (
```
Comment (ditsuke) [inline src/components/WeatherSuggestions/Suggestions.jsx:32]: So we might have another problem here:

image

I think there are no checks if we fail to find a match for the weather in items.js. For me the weather is drizzle so it fails to match any suggestion (and `items` remains undefined).

Comment (ditsuke) [reply src/components/WeatherSuggestions/Suggestions.jsx:32]: We can probably use a default or not render the component on failure to match.

Comment (ditsuke) [reply src/components/WeatherSuggestions/Suggestions.jsx:38]: Resolved with 4f38b04

**Comment by CIPHERTron** [comment]
Thanks for the fix @ditsuke ✨

**Comment by Chukwuemeka-Mba** [comment]
Blank screen showing after getting this error.

Screen Shot 2022-02-03 at 1 04 24 PM

**Comment by ditsuke** [comment]
@Chukwuemeka-Mba 

> Blank screen showing after getting this error.
> 
> Screen Shot 2022-02-03 at 1 04 24 PM

I think removing your `node_modules` folder and doing `yarn install` again should fix it

**Comment by CIPHERTron** [comment]
Heyy @KshitijDarekar @Dhruv-Sachdev1313 @harshsharma6401 can you check now? The above error is now fixed.
I've checked in my local system and it just works fine 🚀

**Review by KshitijDarekar — APPROVED** [review]

**Review by ditsuke — APPROVED** [review]
Real nice work on this one! LGTM

**Comment by CIPHERTron** [comment]
> Real nice work on this one! LGTM

Thank you for that fix otherwise it wouldn't have been possible 😆 

Merging this since it has got 2 approvals 🚀

**Pull request merged**